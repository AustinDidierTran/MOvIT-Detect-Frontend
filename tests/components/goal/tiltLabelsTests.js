/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin-Didier Tran
 * @author Benjamin Roy
 */

import React from 'react';
import PropTypes from 'prop-types';
import configureMockStore from 'redux-mock-store';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import TiltLabels from '../../../src/components/goal/tiltLabels';

Enzyme.configure({ adapter: new Adapter() });

describe('TiltLabels Tests', () => {
  const initialState = { applicationReducer: { language: 'en' } };
  const mockStore = configureMockStore();
  const store = mockStore(initialState);
  const props = {
    language: 'en',
    title: 'This is a title',
    tiltFrequecy: 60,
    tiltLength: 30,
    tiltAngle: 15,
    modifiable: true,
  };

  it('should have proptypes', () => {
    const actualValue = TiltLabels.WrappedComponent.propTypes;

    const expectedValue = {
      language: PropTypes.string.isRequired,
      tiltFrequecy: PropTypes.number,
      tiltLength: PropTypes.number,
      tiltAngle: PropTypes.number,
    };

    expect(JSON.stringify(actualValue)).toEqual(JSON.stringify(expectedValue));
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(<TiltLabels store={store} {...props} />).dive();

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
