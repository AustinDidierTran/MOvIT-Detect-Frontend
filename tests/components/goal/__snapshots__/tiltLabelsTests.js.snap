// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TiltLabels Tests should match the snapshot 1`] = `
<div
  style={
    Object {
      "padding": "10px",
    }
  }
>
  <Card
    className="col-12"
    footer={null}
    header={
      <div
        className="ui-card-title"
        style={
          Object {
            "marginLeft": "14px",
            "marginTop": "14px",
          }
        }
      >
        This is a title
          
        <i
          className="fa fa-pencil"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        />
      </div>
    }
    id={null}
    style={
      Object {
        "backgroundColor": "white",
        "boxShadow": "5px 5px 5px gainsboro",
        "padding": "5px",
      }
    }
    subtitle={null}
    title={null}
  >
    <div
      className="row"
    >
      <div
        className=" col-12 col-md-4 ml-md-4"
      >
        <div
          className="row"
        >
          <span
            className="ml-3"
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            goals.tiltFrequency.en
          </span>
          <span
            className="ml-3"
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            60
             
            time.min.en
          </span>
        </div>
        <div
          className="row"
        >
          <span
            className="ml-3"
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            goals.tiltLength.en
          </span>
          <span
            className="ml-3"
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            30
             
            time.min.en
          </span>
        </div>
        <div
          className="row"
        >
          <span
            className="ml-3"
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            goals.tiltAngle.en
          </span>
          <span
            className="ml-3"
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            15
             °
          </span>
        </div>
      </div>
      <div
        className="col-2 offset-1"
      >
        <img
          alt="chair"
          height="50"
          src={
            Object {
              "default": "",
            }
          }
          style={
            Object {
              "transform": "rotate(-15deg)",
            }
          }
          width="50"
        />
      </div>
    </div>
  </Card>
</div>
`;
